### PR TITLE
fix(material/tabs): add header divider for m3

### PR DESCRIPTION
**Pull request by andrewseguin**

### CODE REVIEW

File: src/material/tabs/_tabs-common.scss [hunk 1]
```diff
@@ -343,6 +343,14 @@ $mat-tab-animation-duration: 500ms !default;
   flex-grow: 1;
   overflow: hidden;
   z-index: 1;
+  border-bottom-style: solid;
+  @include token-utils.use-tokens(
+      tokens-mat-tab-header.$prefix,
```
Comment (crisbeto) [inline src/material/tabs/_tabs-common.scss:348]: nit: seems like a bit too much indentation here.

File: src/material/tabs/_tabs-common.scss [hunk 2]
```diff
@@ -343,6 +343,14 @@ $mat-tab-animation-duration: 500ms !default;
...
+      tokens-mat-tab-header.get-token-slots()
+  ) {
+    @include token-utils.create-token-slot(border-bottom-width, divider-height);
+    @include token-utils.create-token-slot(border-bottom-color, divider-color);
```
Comment (crisbeto) [inline src/material/tabs/_tabs-common.scss:352]: There's an option to position the header at the bottom which we may have to account for: https://material.angular.io/components/tabs/examples#tab-group-header-below

Comment (mmalerba) [reply src/material/tabs/_tabs-common.scss:352]: ooh good catch, PR lgtm otherwise

Comment (andrewseguin) [reply src/material/tabs/_tabs-common.scss:354]: Nice! Added this to the dev-app, I didn't realize this was an option. Thanks for catching this

File: src/material/tabs/_tabs-common.scss (hunk 1 above)
Comment (andrewseguin) [reply src/material/tabs/_tabs-common.scss]: Fixed - thanks. I'd expect formatters to catch this

**Review by mmalerba — APPROVED** [review]

**Pull request merged**